### ISSUE TITLE
Mobile Android - Flickering image after swiping image

### ISSUE THREAD

**Issue by Dev2B** [dimsemenov/PhotoSwipe#989]
Hi, 

thanks for the great tool, Im using it in my mobile webpage. 

One issue I found and its really having an impact on Android user experience. 

Here an example: 
http://goo.gl/8181XB

When swiping between images, let say Image X to Image Y, Image Y shows up shortly then turns to black (i think the image is getting transparent or the background is getting over the image) for some milliseconds and then shows up again correctly. Im testing basically on Android 4.4.4 and 4.2.2 with the latest version of Photoswipe.

I searched the web but no clue. 
Thanks for any suggestion

**Comment by dimsemenov** [comment]
Can you repeat the problem on http://photoswipe.com ?

**Comment by Dev2B** [comment]
Its working nicely on photoswipe.com. 

this is my code actually, note that this is not accuring on web but only on mobile. Thanks for any help 

<?php
<script type="text/javascript"> 
	var openPhotoSwipe = function() {
    var pswpElement = document.querySelectorAll('.pswp')[0];
    var items = [];
	
HTML;
		$totalpages = count($catapages);
		for ( $i=1; $i <= $totalpages; $i++) {
echo <<<HTML
			
			item = {src: '$data$catapath/pages/$i.jpg',w: 0,h: 0}
			items.push(item);
			delete img;
HTML;
		}
		
echo <<<HTML

var options = {modal: false,history: false,focus: true,barsSize : {top:0,bottom:0},captionEl : false,fullscreenEl : false,shareEl : false,closeEl: true,zoomEl: false,bgOpacity :	0.9,tapToClose : false,tapToToggleControls : false, preload: [1,2], loop: false, escKey: false,tapToClose: false,clickToCloseNonZoomable: false,closeOnVerticalDrag: false,closeElClasses:'ui',showAnimationDuration: 0,hideAnimationDuration: 0};

var gallery = new PhotoSwipe( pswpElement, PhotoSwipeUI_Default, items, options);
	
	gallery.listen('gettingData', function(index, item) {
        if (item.w < 1 || item.h < 1) { 
        var img = new Image(); 
        img.onload = function() { 
        item.w = this.width; 
        item.h = this.height; 
           gallery.invalidateCurrItems(); 
           gallery.updateSize(true); 
        }
	
    img.src = item.src; 
    }
	});
	
    gallery.init();
	};
	openPhotoSwipe();
</script>

HTML;

?>

**Comment by Dev2B** [comment]
anyone who has the same problem ?

Thank you

**Comment by MicDeBug** [comment]
Same Problem here.
Using it with Cordova InAppBrowser on Android.

**Comment by Dev2B** [comment]
Theres 2 other issues that I think related to this one: On Android mobile (for all browsers that i ve so far tested, chrome, firefox, stockbrowser), 

1- when swipping fast in the same direction, images are shown first then flickering then disappearing, swipping back normally (not fast) will bring then the same issue: image appears, flick then disappears    

2- when swipping fast in the same direction (ex right), images shown are randomly not shown according to the count order mentionned above, swipping back and then forward gets the right image 

http://goo.gl/8181XB

any clue ?

**Comment by necfol** [comment]
i have the same problem

**Comment by meiyzeng** [comment]
i have the same problem， too

**Comment by emwee** [comment]
same issue here, but on iPhone. I more or less use the same technique as Dev2B (as I don't know the image dimensions on forehand).

**Comment by Createstandard** [comment]
I think the first issue is related to this hack: https://github.com/dimsemenov/PhotoSwipe/issues/796 I think it is just a redraw and you can't see it on desktop but it's still there. I think in your case it's cause you are adding the dimension in later? Essentially I think the plugin is rendering the images, then redrawing them with new dimensions, causing an issue sort of like Flash of unstyled content: https://en.wikipedia.org/wiki/Flash_of_unstyled_content

#2 from your list is a more serious issue in my opinion and is happening regularly to me when using large images also. I have not stepped through the code but it seems like it is listening for the first image to load before starting on the next one and if you get to the next one and the first one hasn't loaded, it freaks out.

**Comment by cindyliu-yb** [comment]
Hello, I have the same problem on ios and android.  Any update on this? thanks

**Comment by Pl-acid** [comment]
Same problem here. I figured out that my problem was caused by the unnecessary updateSize() call
on the items that had already updated their sizes. Here is my solution:
```
gallery.listen('gettingData', function(index, item) {
    if (item.w < 1 || item.h < 1) { 
        var img = new Image(); 
        img.onload = function() { 
            item.w = this.width; 
            item.h = this.height; 
            item.needsUpdate = true //only update the newly loaded item
            gallery.updateSize(true); 
        }

        img.src = item.src; 
    }
});
```

It works fine with v4.0.8. Hope this help.

**Comment by dimsemenov** [comment]
fixed by v5.0
